src/gen_rpc_helper.erl

```diff
@@ -41,11 +43,35 @@
          get_control_receive_timeout/0,
          get_inactivity_timeout/1,
          get_async_call_inactivity_timeout/0,
-         get_listen_ip_config/0]).
+         get_listen_ip_config/0,
+         term_to_iovec/1]).
 
 %%% ===================================================
 %%% Public API
 %%% ===================================================
+
+%% term_to_iovec/1 wrapper to conditionally compress based on threshold
+-spec term_to_iovec(term()) -> ext_iovec.
+term_to_iovec(Term) ->
+  {ok, Compress} = application:get_env(?APP, compress),
+  do_term_to_iovec(Term, Compress).
+
+do_term_to_iovec(Term, Compress) when is_integer(Compress), Compress > 0 ->
+    Size = erlang:external_size(Term),
+    {ok, CompressionThreshold} = application:get_env(?APP, compression_threshold),
+    case Size > CompressionThreshold of
+        true  ->
+            Data = erlang:term_to_iovec(Term, [{compressed, Compress}]),
+            ?tp_ignore_side_effects_in_prod(gen_rpc_compress_payload, #{ threshold => CompressionThreshold,
```

Comment:
@filipecabaco that include allows using this `?tp_..` to trace something. That's from this library: https://github.com/kafka4beam/snabbkaffe

Reply:
👍